pkg/api/addtask.go

```diff
@@ -15,30 +15,61 @@ import (
 
 // Функция проверки и корректировки даты
 func checkDate(task *db.Task) error {
-	now := time.Now()
+	now := time.Now().In(time.UTC) // Приводим текущее время к UTC
 
+	// Если дата не указана, устанавливаем текущую
 	if task.Date == "" {
-		task.Date = now.Format(nextdate.TimeFormat) // если дата не указана — текущая
+		task.Date = now.Format(nextdate.TimeFormat)
+		log.Printf("Дата не указана, установлена текущая дата: %v", task.Date)
 	}
 
+	// Парсим дату задачи
 	t, err := time.Parse(nextdate.TimeFormat, task.Date)
 	if err != nil {
 		return fmt.Errorf("некорректный формат даты: %v", err)
```

Comment:
При оборачивании нужно использовать `%w` (wrap), а просто при выводе в консоль `%v`
```suggestion
		return fmt.Errorf("некорректный формат даты: %w", err)
```